packages/opentelemetry-tracing/src/Span.ts

```diff
@@ -95,13 +95,37 @@ export class Span implements types.Span, ReadableSpan {
     return this;
   }
 
-  addEvent(name: string, attributes?: types.Attributes): this {
+  /**
+   *
+   * @param name Span Name
+   * @param [attributes] Span attributes or start time
+   * if type is {@type TimeInput} and 3rd param is undefined
+   * @param [startTime] Specified start time for the event
+   */
+  addEvent(
+    name: string,
+    attributes?: types.Attributes | types.TimeInput,
+    startTime?: types.TimeInput
+  ): this {
     if (this._isSpanEnded()) return this;
     if (this.events.length >= this._traceParams.numberOfEventsPerSpan!) {
       this._logger.warn('Dropping extra events.');
       this.events.shift();
     }
-    this.events.push({ name, attributes, time: hrTime() });
+    if (
+      Array.isArray(attributes) ||
```

Comment:
What would you think about making a new function `isTimeInput` and adding it to [time.ts](https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-core/src/common/time.ts)?

I could see us wanting to use this same logic later on. Also, I think here it's missing the tuple time case.